### DISCUSSION TITLE
Manual upgrade Drain node?

### DISCUSSION THREAD

**Discussion by frit0-rb** [rancher/rke2#7165] [Q&A]
Good afternoon,

Normally, via doc (https://docs.rke2.io/upgrade/manual_upgrade#upgrade-rke2-using-the-installation-script), when we want to update manually our RKE2 cluster we normally not drain the nodes because in the doc doesn't appear that info.

it is necessary to drain nodes? it more efficient?

**Comment by brandond** [comment]
If you don't want your workload pods running on the node when RKE2 is restarted to apply the upgrade, then you would want to drain first. If you are OK with them remaining running, and taking a potential short outage as things restart to apply the upgrade, then you don't need to drain.

**Reply by frit0-rb** [reply]
> If you don't want your workload pods running on the node when RKE2 is restarted to apply the upgrade, then you would want to drain first. If you are OK with them remaining running, and taking a potential short outage as things restart to apply the upgrade, then you don't need to drain.

But in our case, if the moment we restart the volumes attached in that worker are lost and Longhorn need to recreate it, What is best way do it? Drain?

**Reply by brandond** [reply]
That sounds like a LH question.
